File: 프로그래머스/unrated/181915. 글자 이어 붙여 문자열 만들기/글자 이어 붙여 문자열 만들기.cs
using System;
using System.Text;

public class Solution {
    public string solution(string my_string, int[] index_list) 
    {
        var sb = new StringBuilder();
        
        foreach(int n in index_list)
            sb.Append(my_string[n]);
        
        return sb.ToString();
    }
}

File: 프로그래머스/0/181915. 글자 이어 붙여 문자열 만들기/글자 이어 붙여 문자열 만들기.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

string solution(string my_string, vector<int> index_list) {
    
    std::ostringstream os;
    
    for (int i : index_list)
        os << my_string[i];
    
    return os.str();
}